Data/ByteString/Lazy/ReadInt.hs

```diff
@@ -0,0 +1,380 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilyDependencies #-}
+
+-- We need @AllowAmbiguousTypes! in order to be able to use @TypeApplications@
+-- to disambiguate the desiered instance of class methods whose instance cannot
+-- be inferred from the caller's context.  We would otherwise have to use
+-- proxy arguments.  In particular this is used for the Word64-valued upper
+-- and lower-bounds of various integral types ('wmax64', 'imax64' and 'imin64').
```

Comment:
```suggestion
-- We need @AllowAmbiguousTypes@ in order to be able to use @TypeApplications@
-- to disambiguate the desired instance of class methods whose instance cannot
-- be inferred from the caller's context.  We would otherwise have to use
-- proxy arguments.  In particular this is used for the Word64-valued upper
-- and lower-bounds of various integral types ('wmax64', 'imax64' and 'imin64').
```